coderd/httpmw/workspaceagent.go

```diff
@@ -30,6 +32,7 @@ func ExtractWorkspaceAgent(db database.Store) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
 			ctx := r.Context()
+			systemCtx := dbauthz.WithAuthorizeSystemContext(ctx, rbac.RolesAdminSystem())
```

Comment:
Hmm, I thought we were getting rid of this?

Reply:
I discussed this with @spikecurtis this morning and he made the point that keeping the actor available in the context even for the system user will be useful down the line for audit logging purposes. The actual change itself is comparatively and can be made in a separate PR if required.

Reply:
I will say that this doesn't make a lot of sense to me as a reader of this code... gaining a privileged context that when passed in to a database function bypasses the authorization is *very* abstract, and really hard to track down what's actually going on.

I'd rather we just pass a `privilegedDB` object around that references the PG database directly without the authz wrapper, this just seems really confusing to me.

Reply:
I'm very strongly against passing the underlying `database.Store` object around.

The way to think about this is that

a) we always have a "principal" for every action
b) we always do an authorization check

So, the privileged context doesn't _turn off or bypass_ authentication, it changes the principal to a privileged one, and that principal always _passes_ authentication checks.  This maintains the same set of abstractions as other callers to the `Store`.

It is important to ensure that we always have a principal to attribute the action to, because the next security project after we get authz working is to also push audit logging down into the persistence layer.  Dropping audit logs in API handlers suffers the exact same problems as doing authz checks there, and audit logs that are not 100% complete are much less useful.

Reply:
From my understanding, a `system` user isn't an actor at all, it's fair to say that it holds a similar level of abstract context-dropping behavior as executing the database directly. If the logs showed me that the `system` deleted a workspace, the only information I'd meaningfully retrieve is the timestamp it occurred at, but nothing else is of value there.

I believe the connected user should authorize all actions and that an initiator or actor could be the system. To me, authorizing through the system is just an abstract way to access the underlying database, which, even if audited, I'm not sure what it provides the user apart from the awareness that _something_ did _something_ at X time.

Reply:
The challenging thing about this PR is we **have to use ctx** to indicate "system user/action" because it is a `db.Store`. 

The alternate idea of `.Sudo()` returning a privileged db requires addeding `.Sudo()` to `database.Store` which doesn't make much sense? 

In that if we do the `.Sudo()` idea, we probably want 2 different interfaces. One being an `AuthzStore` which is a superset of `db.Store`. Or something like this.

```
type AuthzStore interface{
  database.Store
  Sudo() database.Store
}
```

------

Until this is the way we do things, it would be a bit awkward to have code call `Sudo` on the underlying database type when the underlying type is just a database. We'd probably need to make a `type NoopAuthzQuerier struct`??